cfme/tests/openstack/cloud/test_flavors.py

```diff
@@ -0,0 +1,142 @@
+"""Tests for Openstack cloud Flavors"""
+
+import fauxfactory
+import pytest
+
+from cfme.cloud.instance.openstack import OpenStackInstance
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.appliance.implementations.ui import navigator, navigate_to
+from cfme.utils.version import current_version
+from cfme.utils.wait import wait_for
+from widgetastic.utils import partial_match
+
+pytestmark = [
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([OpenStackProvider], scope='function')
+]
+
+
+DISK_SIZE = 1
+RAM = 64
+VCPUS = 1
+SWAP_SIZE = 1
+RXTX = 1
+ZERO_DISK_SIZE = 0
+
+
+@pytest.fixture(scope='function')
+def flavor(appliance, provider):
+    flavor = appliance.collections.cloud_flavors.create(name=fauxfactory.gen_alpha(),
+                               provider=provider,
+                               ram=RAM,
+                               vcpus=VCPUS,
+                               disk=DISK_SIZE,
+                               swap=SWAP_SIZE,
+                               rxtx=RXTX)
+    yield flavor
+
+    if flavor.exists:
+        flavor.delete(wait=False)
+
+
+@pytest.fixture(scope='function')
+def zero_disk_flavor(provider):
+    api_flv = provider.mgmt.api.flavors.create(fauxfactory.gen_alpha(), RAM, VCPUS, ZERO_DISK_SIZE)
+    wait_for(provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
+    zero_disk_flavor = provider.appliance.collections.cloud_flavors.instantiate(
+        api_flv.name, provider)
+
+    yield zero_disk_flavor
+
+    if zero_disk_flavor.exists:
+        zero_disk_flavor.delete(wait=False)
+
+
+@pytest.fixture(scope='function')
+def new_instance(provider, zero_disk_flavor):
+    flavor_name = zero_disk_flavor.name
+    prov_data = provider.data['provisioning']
+    prov_form_data = {
+        'request': {'email': fauxfactory.gen_email(),
+                    'first_name': fauxfactory.gen_alpha(),
+                    'last_name': fauxfactory.gen_alpha()},
+        'catalog': {'num_vms': '1',
+                    'vm_name': fauxfactory.gen_alpha()},
+        'environment': {'cloud_network': prov_data['cloud_network'],
+                        'cloud_tenant': prov_data['cloud_tenant']},
+        'properties': {'instance_type': partial_match(flavor_name)},
+    }
+
+    instance_name = prov_form_data['catalog']['vm_name']
+
+    try:
+        instance = provider.appliance.collections.cloud_instances.create(
+            instance_name,
+            provider,
+            prov_form_data, find_in_cfme=True
+        )
+
+    except KeyError:
+        # some yaml value wasn't found
+        pytest.skip('Unable to find an image map in provider "{}" provisioning data: {}'
+                    .format(provider, prov_data))
+
+    yield instance
+    try:
+        instance.cleanup_on_provider()
```

Comment:
No need for try/except here, cleanup_on_provider checks if it exists, and if it fails to delete we want to see the teardown failure.